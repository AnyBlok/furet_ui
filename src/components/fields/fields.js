export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
