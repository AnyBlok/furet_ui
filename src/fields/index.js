/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
