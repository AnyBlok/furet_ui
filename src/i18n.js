/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
