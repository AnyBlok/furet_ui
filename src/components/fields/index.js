import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
