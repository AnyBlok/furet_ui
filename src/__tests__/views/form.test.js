/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import chai from 'chai';
import Vue from 'vue';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {FormView, addNewDataId, openModeDataId, closeModeDataId, cancelModeDataId,
        X2MFormView, addNewX2MDataId, updateValueX2M, deleteDataX2MDataId} from '../../views/form';
import '../../views/search';
import '../../fields';
import '../../space';

jest.mock('../../server-call')

const views = [
    {
        viewId: '1',
        viewType: 'Form',
    },
    {
        viewId: '2',
        viewType: 'Form',
    },
    {
        viewId: '3',
        viewType: 'Form',
        unclickable: true,
    },
    {
        viewId: '4',
        viewType: 'Unknown',
    },
];

const view_form = {
    viewId: '1',
    viewType: 'Form',
    label: 'View : 3',
    creatable: true,
    deletable: true,
    editable: true,
    onClose: '3',
    model: 'Test',
    template: `
        <div class="columns is-multiline is-mobile">
            <div class="column is-4">
                <furet-ui-form-field-integer
                    v-bind:config="config"
                    name="id"
                    label="ID"
                    required="1"
                    readonly="1"
                />
            </div>
            <div class="column is-8">
                <furet-ui-form-field-string
                    v-bind:config="config"
                    required="fields.number"
                    tooltip="Plop"
                    name="name"
                    label="Label"
                    icon="envelope"
                />
            </div>
            <div class="column is-6">
                <furet-ui-form-field-selection
                    v-bind:config="config"
                    name="state"
                    label="State"
                    v-bind:selections="{'new': 'New', 'started': 'Started', 'done': 'Done'}"
                />
            </div>
            <div class="column is-6">
                <furet-ui-form-field-boolean
                    v-bind:config="config"
                    name="bool"
                    label="Boolean"
                />
            </div>
        </div>
    `,
    buttons: [],
    fields: ["id", "name", "bool", "state"],
}

const data = {
    id: 1,
    name: 'Label 1',
    state: 'new',
    bool: true,
};

const change = {
    id: 1,
    name: 'Label 1 bis',
    state: 'new',
    bool: false,
};

describe('furet-ui-form-view component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FormView, {props: {
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FormView, {props: {
                view: view_form,
                dataId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with data 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FormView, {props: {
                view: view_form,
                dataId: '1',
                data,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with data 3', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FormView, {props: {
                view: view_form,
                dataId: '1',
                data,
                change,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('addNewDataId 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        addNewDataId({
            view: view_form,
            $router: router,
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '1',
        });
        expect(vm.$route.name).toBe('space_menu_action_view_dataId');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '1',
            dataId: vm.$route.params.dataId,
            mode: 'new',
        });
    });
    it('addNewDataId 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        addNewDataId({
            view: view_form,
            $router: router,
            spaceId: '1',
            actionId: '3',
            viewId: '1',
        });
        expect(vm.$route.name).toBe('space_action_view_dataId');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: undefined,
            actionId: '3',
            viewId: '1',
            dataId: vm.$route.params.dataId,
            mode: 'new',
        });
    });
    it('openModeDataId 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        openModeDataId({
            $router: router,
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '4',
            dataId: '5',
        });
        expect(vm.$route.name).toBe('space_menu_action_view_dataId');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '4',
            dataId: '5',
            mode: 'readwrite',
        });
    });
    it('openModeDataId 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        openModeDataId({
            $router: router,
            spaceId: '1',
            actionId: '3',
            viewId: '4',
            dataId: '5',
        });
        expect(vm.$route.name).toBe('space_action_view_dataId');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: undefined,
            actionId: '3',
            viewId: '4',
            dataId: '5',
            mode: 'readwrite',
        });
    });
    it('closeModeDataId 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        closeModeDataId({
            view: view_form,
            $router: router,
            spaceId: '1',
            menuId: '2',
            actionId: '3',
        });
        expect(vm.$route.name).toBe('space_menu_action_view');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '3',
        });
    });
    it('closeModeDataId 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        closeModeDataId({
            view: view_form,
            $router: router,
            spaceId: '1',
            actionId: '3',
        });
        expect(vm.$route.name).toBe('space_action_view');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: undefined,
            actionId: '3',
            viewId: '3',
        });
    });
    it('cancelModeDataId 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        store.commit('UPDATE_CHANGE', {model: view_form.model, dataId: '1', fieldname: 'test', value: 'Test'});
        expect(vm.$route.path).toBe('/');
        expect(vm.$store.state.data.changes.Test['1'].test).toBe('Test');
        cancelModeDataId({
            view: view_form,
            $router: router,
            $store: store,
            mode: 'new',
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            dataId: '1',
        });
        expect(vm.$route.name).toBe('space_menu_action_view');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: '2',
            actionId: '3',
            viewId: '3',
        });
        chai.expect(vm.$store.state.data.changes).to.deep.equal({});
    });
    it('cancelModeDataId 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        router.push({path: '/'});
        expect(vm.$route.path).toBe('/');
        store.commit('UPDATE_CHANGE', {model: view_form.model, dataId: '1', fieldname: 'test', value: 'Test'});
        expect(vm.$store.state.data.changes.Test['1'].test).toBe('Test');
        cancelModeDataId({
            view: view_form,
            $router: router,
            $store: store,
            mode: 'readwrite',
            spaceId: '1',
            actionId: '3',
            viewId: '1',
            dataId: '1',
        });
        expect(vm.$route.name).toBe('space_action_view_dataId');
        chai.expect(vm.$route.params).to.deep.equal({
            spaceId: '1',
            menuId: undefined,
            actionId: '3',
            viewId: '1',
            dataId: '1',
            mode: 'readonly',
        });
        chai.expect(vm.$store.state.data.changes).to.deep.equal({});
    });
});

describe('furet-ui-x2m-form-view component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render without data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(X2MFormView, {props: {
                view: view_form,
                views: [],
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(X2MFormView, {props: {
                view: view_form,
                views,
                dataId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(X2MFormView, {props: {
                view: view_form,
                views,
                viewId: '1',
                dataId: '1',
                dataIds: ['1', '2', '3'],
                model: 'Test',
                data,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with data 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(X2MFormView, {props: {
                view: view_form,
                views,
                viewId: '1',
                dataId: '1',
                model: 'Test',
                data,
                change,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('addNewX2MDataId', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        addNewX2MDataId({
            view: view_form,
            dataIds: ['1', '2', '3'],
            viewId: '3',
            x2oField: 'x2oField',
            x2oFieldId: '1',
            $emit: (key, value) => {
                switch (key) {
                    case 'changeView':
                        expect(value).toBe('3');
                        break;
                    case 'updateDataIds':
                        expect(value.length).toBe(4);
                        break;
                    default:
                        expect('Bad key').toBe(1);
                }
            },
            updateValueX2M: (newId, values, bool) => {
                expect(values.dataId).toBe(newId);
                expect(values.x2oField).toBe('1');
                expect(bool).toBe(true);
            },
        });
    });
    it('deleteDataX2MDataId', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        deleteDataX2MDataId({
            view: view_form,
            dataIds: ['1', '2', '3'],
            dataId: '3',
            viewId: '3',
            $emit: (key, value) => {
                switch (key) {
                    case 'updateDataIds':
                        chai.expect(value).to.deep.equal(['1', '2']);
                        break;
                    case 'changeView':
                        expect(value).toBe('3');
                        break;
                    default:
                        expect('undefined key').toBe(key)
                };
            },
            $store: store,
        });
        expect(store.state.data.changes.Test['3']).toBe('DELETED');
    });
    it('updateValueX2M', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            template: '<div />',
        });
        updateValueX2M({
            view: view_form,
            $store: store,
        }, '1', {'x2oField': 'x2oFieldId'}, true);
        chai.expect(store.state.data.changes.new.Test['1']).to.deep.equal({x2oField: 'x2oFieldId'});
    });
});
