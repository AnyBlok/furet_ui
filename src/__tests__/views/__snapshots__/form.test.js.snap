// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`furet-ui-form-view component Render with data 1 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <!---->
                <!---->
                <!---->
                <!---->
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-floppy-o"></i></span> <span>Save</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <!----> <!----></a></p>
                <!---->
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:20px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span>  </span>
                <p class="help is-danger">this field is required</p>
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-form-view component Render with data 2 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <!---->
                <!---->
                <!---->
                <!---->
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-floppy-o"></i></span> <span>Save</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <!----> <!----></a></p>
                <!---->
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:20px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span> 1 </span>
                <!---->
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="Label 1" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-form-view component Render with data 3 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <!---->
                <!---->
                <!---->
                <!---->
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-floppy-o"></i></span> <span>Save</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <!----> <!----></a></p>
                <!---->
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:20px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span> 1 </span>
                <!---->
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="Label 1" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-form-view component Render without data 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <!---->
                <!---->
                <!---->
                <!---->
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-floppy-o"></i></span> <span>Save</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <!----> <!----></a></p>
                <!---->
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:20px;">
        <div config="[object Object]"></div>
    </section>
</div>
`;

exports[`furet-ui-x2m-form-view component Render with data 1 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-plus"></i></span> <span>Create</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-trash"></i></span> <span>Delete</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <span>Close</span></a></p>
            </div>
        </div>
        <div class="level-right">
            <div class="field has-addons">
                <p class="control"><span class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <!---->
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:10px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span> 1 </span>
                <!---->
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="Label 1" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-x2m-form-view component Render with data 2 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-plus"></i></span> <span>Create</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-trash"></i></span> <span>Delete</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <span>Close</span></a></p>
            </div>
        </div>
        <div class="level-right">
            <div class="field has-addons">
                <p class="control"><span class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <!---->
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:10px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span> 1 </span>
                <!---->
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="Label 1" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true" checked="checked"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-x2m-form-view component Render without data 1 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-plus"></i></span> <span>Create</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-trash"></i></span> <span>Delete</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <span>Close</span></a></p>
            </div>
        </div>
        <div class="level-right">
            <!---->
        </div>
    </nav>
    <section class="box" style="margin-bottom:10px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span>  </span>
                <p class="help is-danger">this field is required</p>
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;

exports[`furet-ui-x2m-form-view component Render without data 2 1`] = `
<div data-server-rendered="true">
    <nav class="level">
        <div class="level-left">
            <div class="field has-addons">
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-plus"></i></span> <span>Create</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-trash"></i></span> <span>Delete</span></a></p>
                <p class="control"><a class="button"><span class="icon is-small"><i class="fa fa-times"></i></span> <span>Close</span></a></p>
            </div>
        </div>
        <div class="level-right">
            <div class="field has-addons">
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
                <!---->
                <p class="control"><span class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><!----></span></a>
                    </span>
                </p>
            </div>
        </div>
    </nav>
    <section class="box" style="margin-bottom:10px;">
        <div class="columns is-multiline is-mobile">
            <div class="column is-4"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">ID</label> <span>  </span>
                <p class="help is-danger">this field is required</p>
            </div>
            </span>
        </div>
        <div class="column is-8"><span data-label="Plop" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">Label</label> <div class="control has-icons-left is-clearfix"><input type="text" autocomplete="on" value="" class="input"> <span class="icon is-left"><i class="fa fa-envelope"></i></span>
            <!---->
            <!---->
        </div>
        <!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">State</label> <div class="control is-expanded"><span class="select is-fullwidth"><select><!----> <option value="new">
                        New
                    </option><option value="started">
                        Started
                    </option><option value="done">
                        Done
                    </option></select></span>
    <!---->
</div>
<!---->
</div>
</span>
</div>
<div class="column is-6"><span data-label="" class="tooltip is-primary is-top is-medium"><label tabindex="0" class="checkbox"><input type="checkbox" true-value="true"> <span class="check"></span> <span class="control-label">
                Boolean
            </span></label>
    </span>
</div>
</div>
</section>
</div>
`;
